egs/wsj/s5/steps/pytorchnn/compute_sentence_scores.py

```diff
@@ -0,0 +1,202 @@
+# Copyright 2020    Ke Li
+
+""" This script computes sentence scores with a PyTorch trained neural LM.
+    It is called by steps/pytorchnn/lmrescore_nbest_pytorchnn.sh
+"""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import os
+import argparse
+from collections import defaultdict
+
+import torch
+import torch.nn as nn
+
+
+def load_nbest(path):
+    """Read nbest list into a dictionary.
+
+    Assume the file format is as following:
+    en_4156-A_030185-030248-1 oh yeah
+    en_4156-A_030470-030672-1 well i'm going to have mine and two more classes
+    en_4156-A_030470-030672-2 well i'm gonna have mine and two more classes
+    ...
+    """
+    nbest = defaultdict()
+    with open(path, 'r', encoding='utf-8') as f:
+        for line in f:
+            line = line.strip()
+            try:
+                key, hyp = line.split(' ', 1)
+            except ValueError:
+                key = line
+                hyp = ' '
+            key = key.rsplit('-', 1)[0]
+            if key not in nbest:
+                nbest[key] = [hyp]
+            else:
+                nbest[key].append(hyp)
+    return nbest
+
+
+def read_vocab(path):
+    word2idx = {}
+    idx2word = []
+    with open(path, 'r', encoding='utf-8') as f:
+        for line in f:
+            word = line.split()
+            assert len(word) == 2
+            word = word[0]
+            if word not in word2idx:
+                idx2word.append(word)
+                word2idx[word] = len(idx2word) - 1
+    return word2idx
+
+
+def get_input_and_target(hyp, vocab):
+    """Given a word hypothesis, convert it to integers as input and target.
```

Comment:
Please document functions much better, which includes describing the parameters and return value.
This function has a strange interface because only hyp[0] is accessed (so why not just pass the
1st element)?
I notice you use the same symbol for `<s>` and `</s>`, is this something we discussed, or that
you got from somewhere?
Suggested documentation, assuming you changed the interface so the input wasn't a list:
```
  Converts a sentence to lists of integers, with targets and inputs separately.
 Params:
   hyp: string   Sentence, with words separated by spaces, e.g. 'hello there'
   vocab:  dict from string to int, e.g. { '<s>':1, 'hello': 2, 'there': 3, 'apple': 4, ... }
  Returns: pair of lists, one with the integerized input sequence, one with the integerized
         output sequence: in this case ( [ 1, 2, 3 ], [ 2, 3, 1 ] ), because the input
       sequence has '<s>' added at the start and the output sequence has '<s>' added
      at the end.
      Words that are not in the vocabulary will be converted to '<unk>' (which is expected to be
     in the vocabulary if there are out-of-vocabulary words).
```
The formatting may not be quite right, I think I was attempting to follow some kind of pep3 guidelines but
may not have it right.

Reply:
Sure I will change the interface. As for the bos and eos, the PyTorch word language model example uses the same symbol. I tried both (the same symbol and two different ones for bos and eos) while observed there is almost no difference on results. So I just kept using the same symbol as the preprocessing with it is simpler. 
I will check the python style guide and document functions better. Thanks!